Arm_Software/Arm_Software.ino

```diff
@@ -0,0 +1,356 @@
+
+    Watchdog.attach(estop);
+    WatchdogTelemetry.attach(telemetry);
+    Watchdog.start(watchdogTimeout);
+    WatchdogTelemetry.start(ROVECOMM_UPDATE_RATE);
+}
+
+void loop()
+{
+    updatePosition();
+    parsePackets();
+    updatePosition();
+    if (closedloopActive)
+    {
+        closedLoop();
+    }
+}
+
+void parsePackets()
+{
+    packet = RoveComm.read();
+
+    switch ( packet.data_id )
+    {
+        case RC_ARMBOARD_ARMVELOCITYCONTROL_DATA_ID:
+            int16_t* motorSpeeds; 
+            motorSpeeds = (int16_t*)packet.data;
+            closedloopActive = false;
+            ShoulderTilt.DriveMotor(motorSpeeds[0]);
+            ShoulderTwist.DriveMotor(motorSpeeds[1]);
+            ElbowTilt.DriveMotor(motorSpeeds[2]);
+            ElbowTwist.DriveMotor(motorSpeeds[3]);
+            Wrist.tiltTwistDrive(motorSpeeds[4], motorSpeeds[5]);
+            Watchdog.clear();
+            break;
+        case RC_ARMBOARD_LASERS_DATA_ID:
+            if (packet.data[0])
+            {
+                digitalWrite(LaserToggle, HIGH);
+            }
+            else
+            {
+                digitalWrite(LaserToggle, LOW);
+            }
+            Watchdog.clear();
+            break;
+        case RC_ARMBOARD_SOLENOID_DATA_ID:
+            if (packet.data[0])
+            {
+                digitalWrite(SolenoidToggle, HIGH);
+            }
+            else
+            {
+                digitalWrite(SolenoidToggle, LOW);
+            }
+            Watchdog.clear();
+            break;
+        case RC_ARMBOARD_ARMMOVETOPOSITION_DATA_ID:
+            int16_t* motorAngles;
+            motorAngles = (int16_t*)packet.data;
+            closedloopActive = true;
+            shoulderTiltTarget = motorAngles[0]; 
+            shoulderTwistTarget = motorAngles[1];
+            elbowTiltTarget = motorAngles[2];
+            elbowTwistTarget = motorAngles[3];
+            wristTiltTarget = motorAngles[4];
+            wristTwistTarget = motorAngles[5];
+            Watchdog.clear();
+            break;
+        case RC_ARMBOARD_GRIPPERMOVE_DATA_ID:
+            int16_t* gripperSpeed;
+            gripperSpeed = (int16_t*)packet.data;
+            Gripper.DriveMotor(gripperSpeed[0]);
+            Watchdog.clear();
+            break;
+        default:
+            break;
+    }   
+}
+
+
+void updatePosition()
+{
+    jointAngles[0] = ShoulderTilt.encoder.readDegrees();
+    jointAngles[1] = ShoulderTwist.encoder.readDegrees();
+    jointAngles[2] = ElbowTilt.encoder.readDegrees();
+    jointAngles[3] = ElbowTwist.encoder.readDegrees();
+    jointAngles[4] = Wrist.tiltEncoder.readDegrees();
+    jointAngles[5] = Wrist.twistEncoder.readDegrees();
+}
+
+void closedLoop()
+{
+    float outputs[2];
+    float angles[2];
+    angles[0] = jointAngles[1];
+    angles[1] = jointAngles[0];
+    updatePosition();
+    moveToAngle(ShoulderTilt, shoulderTiltTarget, angles[0], outputs[0]);
+    float shoulderTilt = outputs[0];
```

Comment:
I'd recommend looking at 2020/2021 arm code to see how closed loop/movetoangle() code can be cleaned up. Might be worth having a conversation about in greater detail.